openvdb/openvdb/Platform.h

```diff
@@ -92,6 +92,29 @@
 #define OPENVDB_UNLIKELY(x) (x)
 #endif
 
+/// Macros for assume builtins. Note that we currently don't simply apply these
+/// in place of asserts (when asserts are disabled) - they should be only be
+/// applied with an assert once profiled
+#ifdef __has_cpp_attribute
+  #if __has_cpp_attribute(assume) >= 202207L
+    #define OPENVDB_ASSUME(...) [[assume(__VA_ARGS__)]]
+  #endif
+#endif
+#ifndef OPENVDB_ASSUME
+  #if defined(__clang__)
+    #define OPENVDB_ASSUME(...)  __builtin_assume(__VA_ARGS__);
+  #elif defined(_MSC_VER)
+    #define OPENVDB_ASSUME(...) __assume(__VA_ARGS__);
+  #elif defined(__GNUC__)
+    #if __GNUC__ >= 13
+      #define OPENVDB_ASSUME(...) __attribute__((__assume__(__VA_ARGS__)))
+    #endif
```

Comment:
How substantial is this assume attribute for the performance improvement of ValueAccessors? GCC13 is still not even in the VFX reference platform. Would something like this possibly be a viable alternative until then:
```
#define OPENVDB_ASSUME(...) if (!(__VA_ARGS__)) { __builtin_unreachable(); }
```